xdsl/traits.py

```diff
@@ -75,3 +76,35 @@ def verify(self, op: Operation) -> None:
             raise VerifyException(
                 f"'{op.name}' must be the last operation in the parent block"
             )
+
+
+class IsolatedFromAbove(OpTrait):
+    """
+    Constraint the conained operations to use only values defined inside this
```

Comment:
typo: conained -> contained

maybe add the link to https://mlir.llvm.org/docs/Traits/#isolatedfromabove in the docstring too although it does not explain the recursive check of regions in children ops.